tests/publiccloud/cloud_netconfig.pm

```diff
@@ -12,77 +12,107 @@
 # Maintainer: qa-c team <qa-c@suse.de>
 
 use Mojo::Base 'publiccloud::basetest';
-use Test::Assert 'assert_equals';
+use Test::Assert qw(assert_equals assert_not_equals);
 use testapi;
 use publiccloud::utils qw(is_azure is_ec2 is_gce);
 
-sub run {
-    my ($self, $args) = @_;
-    my $instance = $args->{my_instance};
-    my $os_version = get_var('VERSION');
-    my $pers_net_rules = '/etc/udev/rules.d/75-persistent-net-generator.rules';
+sub query_metadata {
+    my ($self, $instance, %args) = @_;
+    my $ifNum = $args{ifNum};
+    my $addrCount = $args{addrCount};
 
     # Cloud metadata service API is reachable at local destination
     # 169.254.169.254 in case of all public cloud providers.
     my $pc_meta_api_ip = '169.254.169.254';
 
-    my $cloud_netconfig_svc_status = $instance->ssh_script_run(
-        cmd => 'systemctl status cloud-netconfig.service');
-    if ($cloud_netconfig_svc_status == 0 || $cloud_netconfig_svc_status == 3)
-    {
-        record_info('Success', 'Status of cloud-netconfig.service is OK.');
-    }
-    else {
-        die('cloud-netconfig.service status appears to be incorrect!');
+    my $query_meta_ipv4_cmd = "";
+    if (is_azure()) {
```

Comment:
I would create a function in each provider implimentation called `get_ipv4_from_metadata` instead of this if/else structure

Reply:
I will also later need ipv6. I also need to get address for multiple network interfaces.

Reply:
to test `cloud-netconfig` I have to have 2 network interfaces and also some secondary IP address

Reply:
ok then we need different name for this function but it does not change much in my suggestion  ( except function name which TBH I don't care much )